### PR TITLE
Dev/459 When the exam is ongoing, the student can not access the exam which student is not in the white list

### LINKED ISSUE #459: When the exam is ongoing, the student can not access the exam which student is not in the white list

### PR DESCRIPTION
**Pull request by Wally5077**
resolve: #459

### CODE REVIEW

File: Domain/Academy/src/main/java/tw/waterball/judgegirl/academy/domain/utils/ExamValidationUtil.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ */
+public class ExamValidationUtil {
+
+    public static void onlyExamineeCanAccessTheExam(boolean isStudent, int studentId, Exam exam) {
+        if (isStudent && !exam.hasExaminee(studentId)) {
+            throw new ExamineeOnlyOperationException();
+        }
+    }
+
+    public static void onlyWhitelistIpAddressExamineeCanAccessTheOngoingExam(boolean isStudent, int studentId, IpAddress ipAddress, Exam exam) {
```
Comment (Chaoyuuu) [inline Domain/Academy/src/main/java/tw/waterball/judgegirl/academy/domain/utils/ExamValidationUtil.java:20]: rename to `onlyExamineeWithWhitelistIpCanAccessOngoingExam`

File: Spring-Boot/Spring-Boot-Academy/src/main/java/tw/waterball/judgegirl/springboot/academy/repositories/JpaExamRepository.java [hunk 2]
```diff
@@ -159,11 +159,6 @@ public int countAnswersInQuestion(Question.Id id, int studentId) {
         return jpaAnswerDAO.countAllByExamIdAndProblemIdAndStudentId(id.getExamId(), id.getProblemId(), studentId);
     }
 
-    @Override
-    public boolean isExaminee(int studentId, int examId) {
-        return jpaExamineeDAO.existsById_StudentIdAndId_ExamId(studentId, examId);
-    }
```
Comment (Chaoyuuu) [inline Spring-Boot/Spring-Boot-Academy/src/main/java/tw/waterball/judgegirl/springboot/academy/repositories/JpaExamRepository.java:165]: Why did you remove this method?

File: Spring-Boot/Spring-Boot-Academy/src/test/java/tw/waterball/judgegirl/springboot/academy/controllers/ExamControllerTest.java [hunk 3]
```diff
@@ -368,11 +368,12 @@ void GivenOneExamAndOneQuestionCreated_WhenUpdateTheQuestion_ShouldSucceed() thr
 
     @Test
     void GivenOneExamAndOneQuestionCreated_WhenDeleteTheQuestion_ShouldSucceed() throws Exception {
-        ExamView examView = createExamAndGet(now(), now(), "sample-exam");
-        createQuestion(new CreateQuestionUseCase.Request(examView.getId(), 2, 5, 100, 1));
-        deleteQuestion(examView.getId(), PROBLEM_ID)
+        int examId = createExamAndGet(now(), now(), "sample-exam").id;
+        addExaminee(examId, STUDENT_A_ID);
+        createQuestion(new CreateQuestionUseCase.Request(examId, 2, 5, 100, 1));
+        deleteQuestion(examId, PROBLEM_ID)
```
Comment (Chaoyuuu) [inline Spring-Boot/Spring-Boot-Academy/src/test/java/tw/waterball/judgegirl/springboot/academy/controllers/ExamControllerTest.java:374]: Leave a new line between lines 373 and 374.

**Review by Chaoyuuu — CHANGES_REQUESTED** [review]
Review done.

File: Spring-Boot/Spring-Boot-Academy/src/main/java/tw/waterball/judgegirl/springboot/academy/repositories/JpaExamRepository.java (hunk 2 above)
Comment (Wally5077) [reply Spring-Boot/Spring-Boot-Academy/src/main/java/tw/waterball/judgegirl/springboot/academy/repositories/JpaExamRepository.java:165]: Because we only use the `findExam` method to query an `exam` entity, we can just use `exam` to ensure has the examinee participated in the exam or not.

**Review by Chaoyuuu — APPROVED** [review]
LGTM
